server/tserver/src/main/java/org/apache/accumulo/tserver/log/RecoveryLogsIterator.java

```diff
@@ -20,60 +20,85 @@
 
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.Iterator;
 import java.util.List;
 import java.util.Map.Entry;
 
+import org.apache.accumulo.core.client.Scanner;
+import org.apache.accumulo.core.client.ScannerBase;
+import org.apache.accumulo.core.client.rfile.RFile;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.server.ServerContext;
 import org.apache.accumulo.server.fs.VolumeManager;
+import org.apache.accumulo.server.log.SortedLogState;
+import org.apache.accumulo.tserver.logger.LogEvents;
 import org.apache.accumulo.tserver.logger.LogFileKey;
-import org.apache.accumulo.tserver.logger.LogFileValue;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
 import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.Text;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import com.google.common.collect.Iterators;
-import com.google.common.collect.UnmodifiableIterator;
 
 /**
  * Iterates over multiple sorted recovery logs merging them into a single sorted stream.
  */
-public class RecoveryLogsIterator implements CloseableIterator<Entry<LogFileKey,LogFileValue>> {
+public class RecoveryLogsIterator implements Iterator<Entry<Key,Value>>, AutoCloseable {
```

Comment:
I was using what is returned when reading an RFile using `scanner.iterator()`. How would I iterate over the RFile using LogFileKey and LogFileValue?

Reply:
> I was using what is returned when reading an RFile using scanner.iterator(). How would I iterate over the RFile using LogFileKey and LogFileValue?

Can you point me to the code that does this?

Reply:
> Can you point me to the code that does this?

RecoveryLogsIterator.next() could convert from Key,Value, like the following.  Not sure if there is anything using RecoveryLogsIterator that requires a Key/Value though.

```java
public Entry<Key,Value> next() {
    Entry<Key,Value> e = iter.next();
    return new AbstractMap.SimpleImmutableEntry(e.toKey(), e.toValue());
  }
```

Reply:
I use the RFile API here to read each sorted part:
https://github.com/milleruntime/accumulo/blob/7457fe5e0f753f6c9a1289963075c7caac89b45c/server/tserver/src/main/java/org/apache/accumulo/tserver/log/RecoveryLogsIterator.java#L84-L89

Reply:
I made this change in 6a94527457491cd0ff165122b93bfead6b9de85b. I put a try/catch in the method where the IOException was being thrown because I didn't want to have one for all cases. I am not sure the code is shorter but vaving the conversion only happen in one spot is nice.